test/functional/terminal/highlight_spec.lua

```diff
@@ -269,8 +270,8 @@ it('CursorLine and CursorColumn work in :terminal buffer in Normal mode', functi
     bar foobar{4: }                                       |
                                                       |
   ]])
-  -- TermCursorNC has higher precedence.
-  command('highlight TermCursorNC gui=NONE guibg=Red')
+  -- Search has higher precedence.
+  command('highlight Search gui=NONE guibg=Red')
```

Comment:
@zeertzjq I updated this test to use `Search` instead of `TermCursorNC`. All of the tests pass except for the one on line 264: the highlight group on line 270 is `2` instead of `4`. I am not sure why this is the case because the test on line 254 passes correctly.

Will look at it more later, but if you have any ideas why this test is failing let me know.